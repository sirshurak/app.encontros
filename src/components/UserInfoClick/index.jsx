import React from "react";
import AppContext from "../../contexts/AppContext";

function UserInfoClick({ user, children }) {
  return (
    <AppContext.Consumer>
      {({ changePage }) => (
        <button
          onClick={() => changePage("UsersDetail", user)}
          type="button"
          className="link"
        >
          {children}
        </button>
      )}
    </AppContext.Consumer>
  );
}

export default UserInfoClick;
