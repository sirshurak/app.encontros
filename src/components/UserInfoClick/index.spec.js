import { shallow, mount } from 'enzyme';
import UserInfoClick from './index';
import AppContext from '../../contexts/AppContext';

const user = {
    photo: 'https://picsum.photos/200/?image=26',
    username: 'username',
    city: 'Test City',
    gender: 'Man',
    sexualOrientation: 'Heterosexual',
    age: 27
}

describe('UserInfoClick component', () => {
    it('should match shallow snapshop', () => {
        const component = shallow(<UserInfoClick user={user}/>)
        expect(component).toMatchSnapshot();
    });

    it('should mount and click button', () => {
        const changePage = jest.fn();
        const component = mount(
            <AppContext.Provider value={{ changePage }}>
                <UserInfoClick  user={user}/>
            </AppContext.Provider>
        );
        component.find('button').simulate('click', {});
        expect(changePage).toBeCalled();
    });
});