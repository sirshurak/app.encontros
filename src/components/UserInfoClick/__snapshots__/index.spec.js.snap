// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoClick component should match shallow snapshop 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
